iot/api-client/beta-features/commands/receive/receive.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python sample for connecting to Google Cloud IoT Core via MQTT, using JWT.
+This example connects to Google Cloud IoT Core via MQTT, using a JWT for device
+authentication. After connecting, by default the device publishes 100 messages
+to the device's MQTT topic at a rate of one per second, and then exits.
+Before you run the sample, you must follow the instructions in the README
+for this sample.
+"""
+
+# [START iot_mqtt_includes]
+import argparse
+import datetime
+import os
+import ssl
+import time
+
+import jwt
+import paho.mqtt.client as mqtt
+# [END iot_mqtt_includes]
+
+# The initial backoff time after a disconnection occurs, in seconds.
+minimum_backoff_time = 1
+
+# The maximum backoff time before giving up, in seconds.
+MAXIMUM_BACKOFF_TIME = 32
+
+# Whether to wait with exponential backoff before publishing.
+should_backoff = False
+
+
+# [START iot_mqtt_jwt]
+def create_jwt(project_id, private_key_file, algorithm):
+    """Creates a JWT (https://jwt.io) to establish an MQTT connection.
+        Args:
+         project_id: The cloud project ID this device belongs to
+         private_key_file: A path to a file containing either an RSA256 or
+                 ES256 private key.
+         algorithm: The encryption algorithm to use. Either 'RS256' or 'ES256'
+        Returns:
+            An MQTT generated from the given project_id and private key, which
+            expires in 20 minutes. After 20 minutes, your client will be
+            disconnected, and a new JWT will have to be generated.
+        Raises:
+            ValueError: If the private_key_file does not contain a known key.
+        """
+
+    token = {
+            # The time that the token was issued at
+            'iat': datetime.datetime.utcnow(),
```

Comment:
utcnow() is still a datetime object. Do you need to convert to something else so that it can be JSON-serialized? I guess the `jwt` library knows how to handle this?

Reply:
It does appear the JWT library (or the API endpoint) is doing the right thing with the date object, I could instead print as seconds since epoch but in my experiments, this actually seems to break whatever the library / endpoint is doing.

Reply:
I'm guessing the library is actually string parsing this...